<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Mapping reads and summary stats &mdash; Profiling DGE with 48 replicates 3.0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '3.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Profiling DGE with 48 replicates 3.0 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="index.html">Profiling DGE with 48 replicates 3.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="mapping-reads-and-summary-stats">
<span id="mapping"></span><h1>Mapping reads and summary stats<a class="headerlink" href="#mapping-reads-and-summary-stats" title="Permalink to this headline">¶</a></h1>
</div>
<div class="section" id="build-pileup-pl">
<span id="build-pileup-perldoc"></span><h1>build_pileup.pl<a class="headerlink" href="#build-pileup-pl" title="Permalink to this headline">¶</a></h1>
<p>Build aggregated pileup from all BAM files:</p>
<div class="highlight-python"><pre>build_pileup.pl -datapath=./mydata</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-datapath|d</cite> - Path to the data directory of the experiment. Each sub-directory in this will be treated as a separate condition in the experiment (with the name of the condition matching the name of the directory), and each .bam file in each directory is a replicate for that condition.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>-outpath|p</cite> - Path to the output directory.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-conditions|c</cite> - Comma-separated list of conditions. (Default: <em>WT,Snf2</em>)</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="concat-fastq-pl">
<span id="concat-fastq-perldoc"></span><h1>concat_fastq.pl<a class="headerlink" href="#concat-fastq-pl" title="Permalink to this headline">¶</a></h1>
<p>This script takes the multilane, technical replicate data (as produced by
GenePool) and combines it into biological replicate fastq files. It assumes
the source directory has 96 subdirs (numbered 1-96) and that the first 48
are &#8216;Snf2&#8217; samples and the second 48 are &#8216;WT&#8217; samples. It also assumes that
in each of the number subdirs there are seven technical replicate files,
which need combining into one, biological, replicate.</p>
<p>Output files are put in the destination directory under &#8216;Snf2&#8217; and &#8216;WT&#8217;
subdirs. Filenames are derived from the biological replicate number,
sample and MID. e.g. <em>WT_rep06_MID02_allLanes.fastq</em>:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">concat_fastq</span><span class="o">.</span><span class="n">pl</span> <span class="o">--</span><span class="n">src</span><span class="o">-</span><span class="n">path</span> <span class="o">&lt;</span><span class="n">path</span><span class="o">&gt;</span> <span class="o">--</span><span class="n">dest</span><span class="o">-</span><span class="n">path</span> <span class="o">&lt;</span><span class="n">path</span><span class="o">&gt;</span> <span class="p">[</span><span class="o">--</span><span class="n">version</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">verbose</span><span class="o">|--</span><span class="n">no</span><span class="o">-</span><span class="n">verbose</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">debug</span><span class="o">|--</span><span class="n">no</span><span class="o">-</span><span class="n">debug</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">man</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">help</span><span class="p">]</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;src-path</cite> - Source directory with all replicate data. Must have numeric subdirs with fastq files.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;dest-path</cite> - Destination directory for saving concatenated fastq files</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;version</cite> - Report version info and exit</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default:none]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default:none]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="count-fastq-reads-lanes-pl">
<span id="count-fastq-reads-lanes-perldoc"></span><h1>count_fastq_reads_lanes.pl<a class="headerlink" href="#count-fastq-reads-lanes-pl" title="Permalink to this headline">¶</a></h1>
<p>Counts reads in each condition, replicate and lane. Creates a file
<em>all_readcount_lane.dat</em> with counts. Requires directory with subdirs, one
per biological sample, containing lean FASTQ files. There are some numbers and
names hardcoded for our data, so this simple script needs redoing for other
data sets:</p>
<div class="highlight-python"><pre>count_fastq_reads_lanes.pl -dir=/dir/with/fastq/files</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-dir</cite> - Directory with FASTQ files. File names should be in a specific format, so this very simple script really works only with our experiment.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="count-fastq-reads-pl">
<span id="count-fastq-reads-perldoc"></span><h1>count_fastq_reads.pl<a class="headerlink" href="#count-fastq-reads-pl" title="Permalink to this headline">¶</a></h1>
<p>Counts reads in all fastq files and store results in output files
<em>&lt;cond&gt;_readcount.dat</em>.There are some numbers and names hardcoded for our
data, so this simple script needs redoing for other data sets:</p>
<div class="highlight-python"><pre>count_fastq_reads.pl -dir=/dir/with/fastq/files</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-dir</cite> - Directory with FASTQ files. File names should be in a specific format, so this very simple script really works only with our experiment.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="make-bamlinks-pl">
<span id="make-bamlinks-perldoc"></span><h1>make_bamlinks.pl<a class="headerlink" href="#make-bamlinks-pl" title="Permalink to this headline">¶</a></h1>
<p>Creates symbolic links to .bam files in subdirectories.
<a class="reference internal" href="#run-alignments-perldoc"><em>run_alignements.pl</em></a> and
<a class="reference internal" href="#run-biological-alignments-perldoc"><em>run_biological_alignements.pl</em></a>
scripts create individual directories for each replicate under one top
(condition) directory. Each directory contain C&lt;accepted_hits.bam&gt; file.
However, further analysis requires that all bam files are in one directory.
This script creates symbolic links from the top directory to individual bam
files.</p>
<blockquote>
<div>make_bamlinks.pl -topdir=/cluster/gjb_lab/cdr/GRNAseq/mapping/genome/WT</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-topdir</cite> - Top directory containing bam subdirectories. Typically, this contains one condition.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="pileup-var-pl">
<span id="pileup-var-perldoc"></span><h1>pileup_var.pl<a class="headerlink" href="#pileup-var-pl" title="Permalink to this headline">¶</a></h1>
<p>Create files with pileup statistic for each gene: one file with mean and one
with standard deviation. There is one gene per line, and each line contains
gene name and 48 numbers for all replicates. Requires pileup files for each
chromosome, created with <a class="reference internal" href="#build-pileup-perldoc"><em>build_pileup.pl</em></a>.</p>
<blockquote>
<div>pileup_var.pl</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-pileupdir</cite> - The directory with pileups (created with <a class="reference internal" href="#build-pileup-perldoc"><em>build_pileup.pl</em></a>). If not defined, the default value from <em>defs.dat</em> file will be used.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>-outdir</cite> - The directory to save results. If not specified, it will the pileup directory.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>-gff</cite> - The GTF file to be used. The default value is <em>Saccharomyces_cerevisiae.EF4.68.gtf</em>.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="run-alignments-pl">
<span id="run-alignments-perldoc"></span><h1>run_alignments.pl<a class="headerlink" href="#run-alignments-pl" title="Permalink to this headline">¶</a></h1>
<p>This scripts takes all 96 replicates from the yeast snf2 mutant experiment
and aligns the raw Fastq reads to the genome with <a class="reference external" href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a>.
The <em>&#8211;src</em>  directory must have 96 subdirs numbered 1-96, where 1-48 are the snf2
mutants and 49-96 are the wild-types.</p>
<p>The <em>&#8211;dest</em> directory must have the <em>Snf2/</em> and <em>WT/</em> subdirs where the
<a class="reference external" href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a> runs will be saved:</p>
<div class="highlight-python"><pre>run_alignments.pl --src &lt;dir&gt; --dest &lt;dir&gt; [--genome &lt;pathprefix&gt;] [--verbose|--no-verbose] [--debug|--no-debug] [--man] [--help]</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;src</cite> - Source path of subdirs with fastq files</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;dest</cite> - Destination path for output.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;genome</cite> - Path and to bowtie2 genome index [default: /db/bowtie2/Scerevisiae68_ERCC92]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default:none]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default:none]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="run-biological-alignments-pl">
<span id="run-biological-alignments-perldoc"></span><h1>run_biological_alignments.pl<a class="headerlink" href="#run-biological-alignments-pl" title="Permalink to this headline">¶</a></h1>
<p>Given a set of combined biological replicate fastq files run
<a class="reference external" href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a> on them. Use this
to run <a class="reference external" href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a> over the set
of fastq files where technical replicates have been combined into biological
replicates. For this experiment it means 48 replicates each of WT and
Snfs2 samples:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">run_biological_alignments</span><span class="o">.</span><span class="n">pl</span> <span class="o">--</span><span class="n">src</span><span class="o">-</span><span class="n">path</span> <span class="o">&lt;</span><span class="n">path</span><span class="o">&gt;</span> <span class="o">--</span><span class="n">dest</span><span class="o">-</span><span class="n">path</span> <span class="o">&lt;</span><span class="n">path</span><span class="o">&gt;</span> <span class="p">[</span><span class="o">--</span><span class="n">verbose</span><span class="o">|--</span><span class="n">no</span><span class="o">-</span><span class="n">verbose</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">version</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">debug</span><span class="o">|--</span><span class="n">no</span><span class="o">-</span><span class="n">debug</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">man</span><span class="p">]</span> <span class="p">[</span><span class="o">--</span><span class="n">help</span><span class="p">]</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;src-path</cite> - Path containing &#8216;WT&#8217; and &#8216;Snf2&#8217; subdirs with fastq files.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;dest-path</cite> - Path containing &#8216;WT&#8217; and &#8216;Snf2&#8217; subdirs for saving tophat output.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;version</cite> - Report version information and exit</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default:none]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default:none]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="module-sort_and_index_bams">
<span id="sort-and-index-bams-py"></span><span id="sort-and-index-bams-autodocs"></span><h1>sort_and_index_bams.py<a class="headerlink" href="#module-sort_and_index_bams" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id4">
<h2>sort_and_index_bams.py<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h2>
<p>Given a directory, sorts and then indexes every bam file in the directory
using a <a class="reference external" href="http://www.drmaa.org/">DRMAA</a> enabled cluster.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">version:</th><td class="field-body">1.3</td>
</tr>
</tbody>
</table>
<div class="section" id="command-line-arguments">
<h3>Command-line Arguments<a class="headerlink" href="#command-line-arguments" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt><strong>usage:</strong> </dt>
<dd>sort_and_index_bams.pl 
<em class="xref std std-option">-d|--datapath</em> <em>&lt;path&gt;</em>
<em class="xref std std-option">-l|--logfile</em> <em>&lt;file&gt;</em> 
[<a class="reference internal" href="#cmdoption--samtoolspath"><em class="xref std std-option">--samtoolspath</em></a> <em>&lt;path&gt;</em>]
[<a class="reference internal" href="#cmdoption--clustq"><em class="xref std std-option">--clustq</em></a> <em>&lt;str&gt;</em>] 
[<a class="reference internal" href="#cmdoption--tmpdir"><em class="xref std std-option">--tmpdir</em></a> <em>&lt;path&gt;</em>]
[<em class="xref std std-option">-v|--verbose</em>] 
[<em class="xref std std-option">-h|--help</em>]</dd>
</dl>
<div class="section" id="required-parameters">
<h4>Required Parameters<a class="headerlink" href="#required-parameters" title="Permalink to this headline">¶</a></h4>
<dl class="option">
<dt id="cmdoption-d">
<tt class="descname">-d</tt><tt class="descclassname"> &lt;path&gt;</tt><tt class="descclassname">, </tt><tt class="descname">--datapath</tt><tt class="descclassname"> &lt;path&gt;</tt><a class="headerlink" href="#cmdoption-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Path to the data directory with the bam files to be processed</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-l">
<tt class="descname">-l</tt><tt class="descclassname"> &lt;filename&gt;</tt><tt class="descclassname">, </tt><tt class="descname">--logfile</tt><tt class="descclassname"> &lt;filename&gt;</tt><a class="headerlink" href="#cmdoption-l" title="Permalink to this definition">¶</a></dt>
<dd><p>The name (inc. path) of the log file to write to.</p>
</dd></dl>

<div class="section" id="other-options">
<h5>Other options<a class="headerlink" href="#other-options" title="Permalink to this headline">¶</a></h5>
<dl class="option">
<dt id="cmdoption--samtoolspath">
<tt class="descname">--samtoolspath</tt><tt class="descclassname"> &lt;path&gt; (Default: /local/bin/samtools)</tt><a class="headerlink" href="#cmdoption--samtoolspath" title="Permalink to this definition">¶</a></dt>
<dd><p>Specify the path to the samtools executable.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption--clustq">
<tt class="descname">--clustq</tt><tt class="descclassname"> &lt;str&gt; (Default: 64bit-pri.q)</tt><a class="headerlink" href="#cmdoption--clustq" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the q to use to run cluster jobs.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption--tmpdir">
<tt class="descname">--tmpdir</tt><tt class="descclassname"> &lt;path&gt; (Default: ./.temp)</tt><a class="headerlink" href="#cmdoption--tmpdir" title="Permalink to this definition">¶</a></dt>
<dd><p>The full path of the temporary directory you with to use for processing
intermediate files.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-v">
<tt class="descname">-v</tt><tt class="descclassname"></tt><tt class="descclassname">, </tt><tt class="descname">--verbose</tt><tt class="descclassname"></tt><a class="headerlink" href="#cmdoption-v" title="Permalink to this definition">¶</a></dt>
<dd><p>Turn on verbose logging.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-h">
<tt class="descname">-h</tt><tt class="descclassname"></tt><tt class="descclassname">, </tt><tt class="descname">--help</tt><tt class="descclassname"></tt><a class="headerlink" href="#cmdoption-h" title="Permalink to this definition">¶</a></dt>
<dd><p>Print a basic description of the tool and its options to STDOUT.</p>
</dd></dl>

</div>
</div>
</div>
<div class="section" id="output">
<h3>Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h3>
<p>Sorted bam file (affixed with &#8216;sorted&#8217;) and a bam file index (appended with 
the extension .bai) for every bam file in the directory.</p>
</div>
</div>
<div class="section" id="functions">
<h2>Functions<a class="headerlink" href="#functions" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="sort_and_index_bams.sortOrIndexBam">
<tt class="descclassname">sort_and_index_bams.</tt><tt class="descname">sortOrIndexBam</tt><big>(</big><em>c_session</em>, <em>bam_file</em>, <em>options</em>, <em>mode='sort'</em>, <em>outfile=False</em><big>)</big><a class="reference internal" href="_modules/sort_and_index_bams.html#sortOrIndexBam"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#sort_and_index_bams.sortOrIndexBam" title="Permalink to this definition">¶</a></dt>
<dd><p>Calls a cluster run of samtools sort/index on a bam file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>c_session</strong> &#8211; a <tt class="xref py py-class docutils literal"><span class="pre">drmaa.Session</span></tt> object for talking to the 
cluster.</li>
<li><strong>bam_file</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; filename of bam to sort, or a sorted bam file to index.</li>
<li><strong>options</strong> &#8211; an <tt class="xref py py-class docutils literal"><span class="pre">optparse.Values</span></tt> instance.</li>
<li><strong>mode</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; either &#8216;sort&#8217; or &#8216;index&#8217; (default: <em>sort</em>).</li>
<li><strong>outfile</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; if <strong>mode</strong> is <em>sort</em>, specify an output filename.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>The <strong>options</strong> must include the keywords:</p>
<blockquote>
<div><ul class="simple">
<li><em>log</em> - specify the logfile to write to.</li>
<li><em>samtoolspath</em> - the location of samtools to use.</li>
<li><em>tmpdir</em> - the path of the temporary directory to use for processing.</li>
<li><em>clustq</em> - the cluster queue to use (can be <em>None</em>).</li>
<li><em>verbose</em> - toggle on/off verbose logging</li>
</ul>
</div></blockquote>
<p>The function returns the filename of the cluster output file.</p>
</dd></dl>

<dl class="function">
<dt id="sort_and_index_bams.sortOrIndexBams">
<tt class="descclassname">sort_and_index_bams.</tt><tt class="descname">sortOrIndexBams</tt><big>(</big><em>files</em>, <em>options</em>, <em>mode='sort'</em><big>)</big><a class="reference internal" href="_modules/sort_and_index_bams.html#sortOrIndexBams"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#sort_and_index_bams.sortOrIndexBams" title="Permalink to this definition">¶</a></dt>
<dd><p>Sort or index a set of bam files.</p>
<p>This function wraps <a class="reference internal" href="#sort_and_index_bams.sortOrIndexBam" title="sort_and_index_bams.sortOrIndexBam"><tt class="xref py py-func docutils literal"><span class="pre">sortOrIndexBam()</span></tt></a>, calling this function 
for each bam file in a directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>files</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#list" title="(in Python v2.7)"><em>list</em></a>) &#8211; a list of bam files to sort, sorted bam files to index.</li>
<li><strong>options</strong> &#8211; an <tt class="xref py py-class docutils literal"><span class="pre">optparse.Values</span></tt> instance.</li>
<li><strong>mode</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; either &#8216;sort&#8217; or &#8216;index&#8217; (default: <em>sort</em>).</li>
<li><strong>outfile</strong> (<a class="reference external" href="http://docs.python.org/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; if <strong>mode</strong> is <em>sort</em>, specify an output filename.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>The <strong>options</strong> must include:</p>
<blockquote>
<div><ul class="simple">
<li><em>log</em> - specify the logfile to write to.</li>
<li><em>samtoolspath</em> - the location of samtools to use.</li>
<li><em>tmpdir</em> - the path of the temporary directory to use for processing.</li>
<li><em>clustq</em> - the cluster queue to use (can be <em>None</em>).</li>
<li><em>verbose</em> - toggle on/off verbose logging</li>
</ul>
</div></blockquote>
<p>The function returns a list of cluster output filenames.</p>
</dd></dl>

</div>
</div>
<div class="section" id="summary-stats-pl">
<span id="summary-stats-perldoc"></span><h1>summary_stats.pl<a class="headerlink" href="#summary-stats-pl" title="Permalink to this headline">¶</a></h1>
<p>Generate summary statistics of alignment infomation. This script generates
simple statistics on the outcome of Tophat alignments. It expects that the
given <em>&#8211;path</em> includes subdirectories generated by tophat which include
<em>accepted_hits.bam</em> and <em>unmapped.bam</em> files in them. Given that assumption
the script with then run &#8216;<a class="reference external" href="http://www.htslib.org/">samtools</a> <em>flagstat</em>&#8216; on
the BAM files to determine the number of mapped, unmapped and failed reads for
each run. These are then reported in the output file:</p>
<div class="highlight-python"><pre>summary_stats.pl --path &lt;path&gt; [--samtools &lt;path&gt;] [--threads &lt;num&gt;] [--out &lt;file&gt;] [--version] [--verbose|--no-verbose] [--debug|--no-debug] [--man] [--help]</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;path</cite> - Path to tophat output dirs.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;samtools</cite> - Path to samtools executeable [default: /sw/samtools-0.1.18/samtools]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;threads</cite> - Number of threads to use [default: 1]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;out</cite> - Output filename. [default: out.tsv]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;version</cite> - Output version info and exit.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default: verbose]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default: no-debug]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="unique-fastq-reads-pl">
<span id="unique-fastq-reads-perldoc"></span><h1>unique_fastq_reads.pl<a class="headerlink" href="#unique-fastq-reads-pl" title="Permalink to this headline">¶</a></h1>
<p>Takes a Fastq file and writes out only unique examples of reads therein.
Keeps header and quality information for the first instance of each read:</p>
<div class="highlight-python"><pre>unique_fastq_reads.pl --in &lt;file&gt; [--out &lt;file&gt;] [--verbose|--no-verbose] [--debug|--no-debug] [--man] [--help]</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;in</cite> - Input fastq file (can be gzipped).</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;out</cite> - Output filename. [default: unique.fastq]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default:none]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default:none]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
<div class="toctree-wrapper compound">
<ul class="simple">
</ul>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Mapping reads and summary stats</a></li>
<li><a class="reference internal" href="#build-pileup-pl">build_pileup.pl</a></li>
<li><a class="reference internal" href="#concat-fastq-pl">concat_fastq.pl</a></li>
<li><a class="reference internal" href="#count-fastq-reads-lanes-pl">count_fastq_reads_lanes.pl</a></li>
<li><a class="reference internal" href="#count-fastq-reads-pl">count_fastq_reads.pl</a></li>
<li><a class="reference internal" href="#make-bamlinks-pl">make_bamlinks.pl</a></li>
<li><a class="reference internal" href="#pileup-var-pl">pileup_var.pl</a></li>
<li><a class="reference internal" href="#run-alignments-pl">run_alignments.pl</a></li>
<li><a class="reference internal" href="#run-biological-alignments-pl">run_biological_alignments.pl</a></li>
<li><a class="reference internal" href="#module-sort_and_index_bams">sort_and_index_bams.py</a><ul>
<li><a class="reference internal" href="#id4">sort_and_index_bams.py</a><ul>
<li><a class="reference internal" href="#command-line-arguments">Command-line Arguments</a><ul>
<li><a class="reference internal" href="#required-parameters">Required Parameters</a><ul>
<li><a class="reference internal" href="#other-options">Other options</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li><a class="reference internal" href="#functions">Functions</a></li>
</ul>
</li>
<li><a class="reference internal" href="#summary-stats-pl">summary_stats.pl</a></li>
<li><a class="reference internal" href="#unique-fastq-reads-pl">unique_fastq_reads.pl</a><ul>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/mapping.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="index.html">Profiling DGE with 48 replicates 3.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2016, N Schurch.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b2.
    </div>
  </body>
</html>