<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>preprocessed_data &amp; data munging scripts &mdash; Profiling DGE with 48 replicates 3.0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '3.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="Profiling DGE with 48 replicates 3.0 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">Profiling DGE with 48 replicates 3.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="preprocessed-data-data-munging-scripts">
<span id="preprocessed-data"></span><h1>preprocessed_data &amp; data munging scripts<a class="headerlink" href="#preprocessed-data-data-munging-scripts" title="Permalink to this headline">¶</a></h1>
<p>This directory contains pre-processed results from several parts of the
processing chain. The aim here is to give people access to the intermediate data
so they can reproduce the paper plots and do their own analysis on the results
without having to run the full processing chain, much of which requires a DRMMA
enabled cluster.</p>
</div>
<div class="section" id="gold-standard-results">
<span id="id1"></span><h1>Gold Standard Results<a class="headerlink" href="#gold-standard-results" title="Permalink to this headline">¶</a></h1>
<p>This sub-directory contains the &#8216;gold standard&#8217; results for each tool and test,
computed using the full set of clean replicates defined in
<a class="reference internal" href="index.html#paper1"><em>Gierlinski et. al. (2015)</em></a>. The files are tab-delimited files
with 11 columns:</p>
<blockquote>
<div><ul class="simple">
<li>GeneID - SGD ID</li>
<li>log2foldchange</li>
<li>significance - adjusted p-value, q-value or FDR depending on the tool</li>
<li>WT - WT counts</li>
<li>Snf2 - Snf2 counts</li>
<li>GeneName - human-friendly gene ID</li>
<li>Description - gene description from ensembl</li>
<li>Chromosome</li>
<li>Start</li>
<li>End</li>
<li>Strand</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="ena-sample-mapping-pl">
<span id="ena-sample-mapping-perldoc"></span><h1>ENA_sample_mapping.pl<a class="headerlink" href="#ena-sample-mapping-pl" title="Permalink to this headline">¶</a></h1>
<p>The metadata associated with the yeast RNA-seq data at ENA is not informative
enough to replicate our sample names. This script generates a more meaningful
mapping for uploading to figshare:</p>
<div class="highlight-python"><pre>ENA_sample_mapping.pl [--in &lt;path&gt;] [--url &lt;ENA url&gt;] [--out &lt;file&gt;] [--verbose|--no-verbose] [--version] [--debug|--no-debug] [--man] [--help]</pre>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;in</cite> - Input path to raw fastq files. [default: /cluster/gjb_lab/cdr/GRNAseq/raw/raw_reads]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;url</cite> - URL generated by ENA for downloading sample metadata file. Only specify if you know what you&#8217;re doing.</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;out</cite> - Output filename. [default: sample_mapping.tsv]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;version</cite> - Report version information and exit</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;verbose|&#8211;no-verbose</cite> - Toggle verbosity. [default:none]</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;debug|&#8211;no-debug</cite> - Toggle debugging output. [default:none]</td>
</tr>
<tr class="field-odd field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;help</cite> - Brief help.</td>
</tr>
<tr class="field-even field"><th class="field-name">param:</th><td class="field-body"><cite>&#8211;man</cite> - Full manpage of program.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="enadata-erp004763-sample-mapping-tsv">
<span id="id2"></span><h1>ENAdata_ERP004763_sample_mapping.tsv<a class="headerlink" href="#enadata-erp004763-sample-mapping-tsv" title="Permalink to this headline">¶</a></h1>
<p>Data file describing the mapping from ENA Run Accession IDs to the sample,
replicate and lane information for the experiment.</p>
</div>
<div class="section" id="wt-countdata-tar-gz-wt-countdata-tar-gz">
<span id="countdata-tar-gz"></span><h1>WT_countdata.tar.gz &amp; WT_countdata.tar.gz<a class="headerlink" href="#wt-countdata-tar-gz-wt-countdata-tar-gz" title="Permalink to this headline">¶</a></h1>
<p>Tar-gzipped gene count data for each biological replicate, computed with
HTSeq-count using the <a class="reference internal" href="bootstrapping.html#group-by-gene-perldoc"><em>group_by_gene.pl</em></a> script.
These files can be used as input for the bootstrapping runs of
<a class="reference internal" href="bootstrapping.html#generic-wrapper-autodoc"><em>generic_wrapper.py</em></a> in place of the bam files.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">preprocessed_data &amp; data munging scripts</a></li>
<li><a class="reference internal" href="#gold-standard-results">Gold Standard Results</a></li>
<li><a class="reference internal" href="#ena-sample-mapping-pl">ENA_sample_mapping.pl</a></li>
<li><a class="reference internal" href="#enadata-erp004763-sample-mapping-tsv">ENAdata_ERP004763_sample_mapping.tsv</a></li>
<li><a class="reference internal" href="#wt-countdata-tar-gz-wt-countdata-tar-gz">WT_countdata.tar.gz &amp; WT_countdata.tar.gz</a></li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/preprocessed_data.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">Profiling DGE with 48 replicates 3.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2016, N Schurch.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b2.
    </div>
  </body>
</html>