.. preprocessed_data:

.. _preprocessed_data:

****************************************
preprocessed_data & data munging scripts
****************************************

This directory contains pre-processed results from several parts of the 
processing chain. The aim here is to give people access to the intermediate data
so they can reproduce the paper plots and do their own analysis on the results
without having to run the full processing chain, much of which requires a DRMMA
enabled cluster.

.. gold_standard_results:

.. _gold_standard_results:

*********************
Gold Standard Results
*********************

This sub-directory contains the 'gold standard' results for each tool and test,
computed using the full set of clean replicates defined in 
:ref:`Gierlinski et. al. (2015) <paper1>`. The files are tab-delimited files 
with 11 columns:

  * GeneID - SGD ID
  * log2foldchange
  * significance - adjusted p-value, q-value or FDR depending on the tool
  * WT - WT counts
  * Snf2 - Snf2 counts
  * GeneName - human-friendly gene ID
  * Description - gene description from ensembl
  * Chromosome
  * Start
  * End
  * Strand

.. ENA_sample_mapping_perldoc:

.. _ENA_sample_mapping_perldoc:

*********************
ENA_sample_mapping.pl
*********************

The metadata associated with the yeast RNA-seq data at ENA is not informative 
enough to replicate our sample names. This script generates a more meaningful 
mapping for uploading to figshare::

	ENA_sample_mapping.pl [--in <path>] [--url <ENA url>] [--out <file>] [--verbose|--no-verbose] [--version] [--debug|--no-debug] [--man] [--help]

:param: `--in` - Input path to raw fastq files. [default: /cluster/gjb_lab/cdr/GRNAseq/raw/raw_reads]

:param: `--url` - URL generated by ENA for downloading sample metadata file. Only specify if you know what you're doing.

:param: `--out` - Output filename. [default: sample_mapping.tsv]

:param: `--version` - Report version information and exit

:param: `--verbose|--no-verbose` - Toggle verbosity. [default:none]

:param: `--debug|--no-debug` - Toggle debugging output. [default:none]

:param: `--help` - Brief help.

:param: `--man` - Full manpage of program.

.. ENAdata_ERP004763_sample_mapping_tsv:

.. _ENAdata_ERP004763_sample_mapping_tsv:

************************************
ENAdata_ERP004763_sample_mapping.tsv
************************************

Data file describing the mapping from ENA Run Accession IDs to the sample, 
replicate and lane information for the experiment.

.. countdata.tar.gz:

.. _countdata.tar.gz:

*****************************************
WT_countdata.tar.gz & WT_countdata.tar.gz
*****************************************

Tar-gzipped gene count data for each biological replicate, computed with 
HTSeq-count using the :ref:`group_by_gene.pl <group_by_gene_perldoc>` script.
These files can be used as input for the bootstrapping runs of 
:ref:`generic_wrapper.py <generic_wrapper_autodoc>` in place of the bam files.
